pkg/appmanager/resourceConfig.go

```diff
@@ -1174,6 +1180,28 @@ func (rc *ResourceConfig) AddRuleToPolicy(
 	rc.SetPolicy(*policy)
 }
 
+func (rc *ResourceConfig) DeleteRuleFromPolicy(
+	policyName string,
+	rule *Rule,
+) {
+	// We currently have at most 1 policy, 'forwarding'
+	policy := rc.FindPolicy("forwarding")
+	if nil != policy {
+		for i, r := range policy.Rules {
+			if r.Name == rule.Name && r.FullURI == rule.FullURI {
+				// Remove old rule
+				if len(policy.Rules) == 1 {
+					rc.DeletePolicy(*policy)
+				} else {
+					policy.Rules = append(policy.Rules[:i], policy.Rules[i+1:]...)
+					rc.SetPolicy(*policy)
```

Comment:
Check out the block here that deletes rules: https://github.com/F5Networks/k8s-bigip-ctlr/blob/master/pkg/appmanager/appManager.go#L1726. You'll have to do something similar with the offsets, and there is a method that specifically deletes rules. (The linked block deletes rules under a different condition, but the method of deleting should be applicable here)